src/twisted/python/compat.py

```diff
@@ -377,13 +399,17 @@ def _matchingString(constantString, inputString):
     @rtype: the type of C{inputString}
     """
     if isinstance(constantString, bytes):
-        otherType = constantString.decode("ascii")
-    else:
-        otherType = constantString.encode("ascii")
-    if type(constantString) == type(inputString):
-        return constantString
+        uniString = constantString.decode("ascii")
+        if isinstance(inputString, bytes):
+            return constantString
+        else:
+            return uniString
     else:
-        return otherType
+        byteString = constantString.encode("ascii")
```

Comment:
Same here. Might get computed and then tossed.

Reply:
(See above)